pkg/secretsmanager/secretsmanager.go

```diff
@@ -409,75 +517,25 @@ func (sm *secretManagerNone) LoadSecret(ctx context.Context, secretName string)
 	return nil, nil
 }
 
-// manageCloudCredentials handles the credential used to access the secret manager
-// credentials are placed in temp files or environmental variables according to the SM specs.
-func manageCloudCredentials(secMan v1alpha1.SecretsManager, secObject *corev1.Secret, dirPath string) error {
-
-	writeFile := func(name string, contents []byte) (string, error) {
-		fPath := path.Join(dirPath, name)
-
-		// Open a new file for writing only
-		file, err := os.OpenFile(
-			fPath,
-			os.O_WRONLY|os.O_TRUNC|os.O_CREATE,
-			0666,
-		)
-		if err != nil {
-			return "", err
-		}
-		defer file.Close()
-
-		// Write bytes to file
-		_, err = file.Write(contents)
-		if err != nil {
-			return "", err
-		}
-		return fPath, nil
+// LoadCredentialsSecret loads the credential secret data from the Kubernetes secret
+func LoadCredentialsSecret(rClient client.Client, config *v1alpha1.AppConfig, cloudCredNS string) (*corev1.Secret, error) {
+	// load credentials secret
+	secObject, err := k8ssecrets.LoadSecret(rClient, config.CredentialsSecretName, cloudCredNS)
+	if err != nil {
+		log.Error(err, "error loading cloud credentials secret from the Kubernetes API",
+			"secret_name", config.CredentialsSecretName,
+			"cloud_secret_namespace", cloudCredNS)
 	}
-	switch secMan {
-	case v1alpha1.SecretsManagerGCP:
-		keyValue, ok := secObject.Data[string(v1alpha1.SecretsManagerGoogleApplicationCredentials)]
-		if !ok {
-			return fmt.Errorf(fmt.Sprintf("%s must be provided in a credentials secret",
-				v1alpha1.SecretsManagerGoogleApplicationCredentials))
-		}
-		fp, err := writeFile("gcp_credentials.json", keyValue)
-		if err != nil {
-			return err
-		}
-		if err := os.Setenv("GOOGLE_APPLICATION_CREDENTIALS", fp); err != nil {
-			return err
-		}
-	case v1alpha1.SecretsManagerAWS:
-		if keyValue, ok := secObject.Data[string(v1alpha1.SecretsManagerAwsAccessKeyID)]; ok {
-			if err := os.Setenv("AWS_ACCESS_KEY_ID", string(keyValue)); err != nil {
-				return err
-			}
-		}
-		if keyValue, ok := secObject.Data[string(v1alpha1.SecretsManagerAwsSecretAccessKey)]; ok {
-			if err := os.Setenv("AWS_SECRET_ACCESS_KEY", string(keyValue)); err != nil {
-				return err
-			}
-		}
-	case v1alpha1.SecretsManagerAzure:
-		if keyValue, ok := secObject.Data[string(v1alpha1.SecretsManagerAzureTenantID)]; ok {
-			if err := os.Setenv("AZURE_TENANT_ID", string(keyValue)); err != nil {
-				return err
-			}
-
-		}
-		if keyValue, ok := secObject.Data[string(v1alpha1.SecretsManagerAzureClientID)]; ok {
-			if err := os.Setenv("AZURE_CLIENT_ID", string(keyValue)); err != nil {
-				return err
-			}
+	return secObject, err
+}
 
-		}
-		if keyValue, ok := secObject.Data[string(v1alpha1.SecretsManagerAzureClientSecret)]; ok {
-			if err := os.Setenv("AZURE_CLIENT_SECRET", string(keyValue)); err != nil {
-				return err
-			}
-		}
+// GetCredentialData extracts the credential data from the data field
+func GetCredentialData(secObject *corev1.Secret) ([]byte, error) {
```

Comment:
You probably don't need to expose this function. 
```suggestion
func getCredentialData(secObject *corev1.Secret) ([]byte, error) {
```

Reply:
done